src/Jackett.Common/Indexers/ShizaProject.cs

```diff
@@ -0,0 +1,202 @@
+            set => configData = value;
+        }
+
+        /// <summary>
+        /// http://shiza-project.com/accounts/login
+        /// </summary>
+        private string LoginUrl => SiteLink + "accounts/login";
+
+        /// <summary>
+        /// https://tr.ShizaProject.com/index.php?do=search
+        /// </summary>
+        private string SearchUrl => SiteLink + "releases/search";
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            var data = new Dictionary<string, string>
+            {
+                { "field-email", Configuration.Username.Value },
+                { "field-password", Configuration.Password.Value }
+            };
+
+            var result = await RequestLoginAndFollowRedirect(
+                LoginUrl,
+                data,
+                CookieHeader,
+                returnCookiesFromFirstCall: true
+            );
+
+            var parser = new HtmlParser();
+            var document = await parser.ParseDocumentAsync(result.ContentString);
+
+            await ConfigureIfOK(result.Cookies, IsAuthorized(result), () =>
+            {
+                const string ErrorSelector = "div.card-content > div.alert-error";
+                var errorMessage = document.QuerySelector(ErrorSelector).Text().Trim();
+                throw new ExceptionWithConfigData(errorMessage, Configuration);
+            });
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        public override async Task<byte[]> Download(Uri link)
+        {
+            await EnsureAuthorized();
+            return await base.Download(link);
+        }
+
+        // If the search string is empty use the latest releases
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query) {
+            WebResult result;
+            if (query.IsTest || string.IsNullOrWhiteSpace(query.SearchTerm)) {
+                result = await RequestWithCookiesAndRetryAsync(SiteLink);
+            } else {
+                // Prepare the search query
+                var queryParameters = new NameValueCollection
+                {
+                    { "q", query.SearchTerm},
+                };
+                result = await RequestWithCookiesAndRetryAsync(SearchUrl + "?" + queryParameters.GetQueryString());
+            }
+
+            const string ReleaseLinksSelector = "article.grid-card > a.card-box";
+            
+            var releases = new List<ReleaseInfo>();
+
+            try
+            {
+                var parser = new HtmlParser();
+                var document = await parser.ParseDocumentAsync(result.ContentString);
+
+                foreach (var linkNode in document.QuerySelectorAll(ReleaseLinksSelector))
+                {
+                    var url = linkNode.GetAttribute("href");
+                    releases.AddRange(await FetchShowReleases(url));
```

Comment:
The only problem I have faced for now, is that when we need to process multiple releases it took more time than it's comfortable.
You can check that if you search empty string.
The problem itself that release page request took 1-2 seconds and they are sequential.

I wonder if someone could consult me how to make pages requests parallel.